addons/sale/tests/test_sale_to_invoice.py

```diff
@@ -210,6 +210,47 @@ def test_downpayment_percentage_tax_icl(self):
         invoice.action_post()
         self.assertEqual(downpayment_line.price_unit, self.sale_order.amount_total/2, 'downpayment should have the correct amount')
 
+    def test_downpayment_invoice_and_partial_credit_note(self):
+        """This test check that the downpayment line amount on the sale order remains consistent"""
+        self.sale_order.action_confirm()
+
+        # Create an invoice for a Down payment of 100
+        payment = self.env['sale.advance.payment.inv'].with_context(self.context).create({
+            'advance_payment_method': 'fixed',
+            'fixed_amount': 100,
+            'deposit_account_id': self.company_data['default_account_revenue'].id,
+        })
+        payment.create_invoices()
+
+        # Ensure the downpayment line on the sale order is correctly set to 100
+        downpayment_line = self.sale_order.order_line.filtered(lambda l: l.is_downpayment)
+        self.assertEqual(downpayment_line.price_unit, 100)
+
+        # post the downpayment invoice and ensure the downpayment_line amount is still 100
+        downpayment_invoice = downpayment_line.order_id.order_line.invoice_lines.move_id
+        downpayment_invoice.action_post()
+        self.assertEqual(downpayment_line.price_unit, 100)
+
+        # Create a credit note for a part of the downpayment invoice and post it
+        move_reversal = self.env['account.move.reversal'].with_context(
+            active_model="account.move",
+            active_ids=downpayment_invoice.ids,
+        ).create({
+            'date': '2020-02-01',
+            'reason': 'no reason',
```

Comment:
I wrote this string years ago and it has been copy pasted so many times now :D

Reply:
It's fun so I get while everyone just copy paste it 😂